etc/cmake/options.cmake

```diff
@@ -230,6 +230,8 @@ ot_option(OT_PING_SENDER OPENTHREAD_CONFIG_PING_SENDER_ENABLE "ping sender" ${OT
 ot_option(OT_PLATFORM_BOOTLOADER_MODE OPENTHREAD_CONFIG_PLATFORM_BOOTLOADER_MODE_ENABLE "platform bootloader mode")
 ot_option(OT_PLATFORM_KEY_REF OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE "platform key reference secure storage")
 ot_option(OT_PLATFORM_NETIF OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE "platform netif")
+ot_option(OT_PLATFORM_POSIX_CONFIG_FILE OPENTHREAD_POSIX_CONFIG_CONFIGURATION_FILE_ENABLE "posix config file")
+ot_option(OT_PLATFORM_POWER_CALIBRATION OPENTHREAD_CONFIG_PLATFORM_POWER_CALIBRATION_ENABLE "power calibration")
```

Comment:
Same about these.  The `option.cmake` is intended for a subset of config option ot OT core.  It would be better not to include platform specific config in this file. I know there are some that were added before like `OT_POSIX_SETTINGS_PATH` but I would sat it may be better to not add more platform config here and  keep the configs separate.

Reply:
`OT_PLATFORM_POSIX_CONFIG_FILE` is a good candidate to move.

What are your thoughts about `OT_PLATFORM_POWER_CALIBRATION`? It looks like `OPENTHREAD_CONFIG_PLATFORM_POWER_CALIBRATION_ENABLE` is also used in `src/lib/spinel` and `src/ncp`. I wasn't sure what the policy was for adding a new `ot_option` was. It looks like this file is the only place they currently exist. Is there a place for configuration options that are shared across different portions of the code?

Reply:
Yes. Thanks. 

`OPENTHREAD_CONFIG_PLATFORM_POWER_CALIBRATION_ENABLE ` is needed within OT core (it will determine whether `mPowerCalibration` variable is included in `ot::Instance` class). So should probably stay in `options.cmake`.

My view is better to move configs to the underlying modules that they belong to (we may need to clean this up later and move some `POSIX` configs that were added in `option.cmake` before).

As general approach, I think it is fine for platform and/or other related modules like cli, ncp, spinel, to know and use the core `OPENTHREAD_CONFIG` but the other way (adding configs for specific platform or CLI or NCP in OT core) may not be good to me